### ISSUE TITLE
Support for Discord Canary?

### ISSUE THREAD

**Issue by 1byte2bytes** [ByzantineFailure/BPM-for-Discord#78]
Is it possible to get Discord Canary support? I would appreciate it a bunch! I looked at implementing it myself and making a pull request, but I don't quite understand how the install system works. I've never used Node before

**Comment by Jibodeah** [comment]
I've been using BPM on Canary since forever and it's always worked for me.

Except when some canary changes break little parts of it.

**Comment by ByzantineFailure** [comment]
@1byte2bytes Do you mean the Linux client or the PTR?  Both are supported.

Please see "PTB Installation" here:
https://github.com/ByzantineFailure/BPM-for-Discord/blob/discord/discord/INSTALLATION.md

**Comment by ByzantineFailure** [comment]
Closing this issue.  If there is some problem with the PTB installers, feel free to reopen it.

**Comment by 1byte2bytes** [comment]
Discord Canary on macOS and Windows are in different folders than PTB. You can't obtain them on the site via the normal downloads page though. Linux can install it fine since you have to manually specify the Discord path.

https://discordapp.com/api/download/canary?platform=win
https://discordapp.com/api/download/canary?platform=osx

**Comment by BytewaveMLP** [comment]
Bump. Especially with video chat being a Canary user feature now, I'd imagine more people will be moving to Canary.

I've tried hacking together my own solution, but with no luck.

**Comment by 1byte2bytes** [comment]
I’ve had success manually replacing PTB with Canary in the paths JS file, but that’s not really a permanent solution. 

Sydney Erickson
https://sydneyerickson.me/

**Comment by BytewaveMLP** [comment]
Could you share your modifications? I tried the same, but with no luck.
I've also noticed that `app.asar` has moved to `original_app.asar`, but maybe that's due to BeautifulDiscord? I'm not sure.

**Comment by 1byte2bytes** [comment]
Yeah, in a bit. Currently in school. 

Sydney Erickson
https://sydneyerickson.me/

**Comment by ByzantineFailure** [comment]
Re-opening.  Super late doing this, but I do intend to look at this at some point when I'm no longer simultaneously working, traveling, and actively interviewing around for a new job (the last few months have been _hellishly_ busy).

**Comment by ByzantineFailure** [comment]
@1byte2bytes if you have those changes could you actually post the diff here?  They'd make a good starting point.

**Comment by 1byte2bytes** [comment]
Yup! My MacBook is software updating at he moment, give me about 10 minutes

**Comment by 1byte2bytes** [comment]
Diff for `discord/installer/lib/paths.js`

```
L48 | -             var localDataFolder = isPTB ? 'DiscordPTB' : 'Discord',
L48 | +             var localDataFolder = isPTB ? 'DiscordCanary' : 'Discord',
```
```
L70 | -             return '/Applications/Discord' + (isPTB ? ' PTB' : '') + '.app/Contents/Resources';
L70 | +             return '/Applications/Discord' + (isPTB ? ' Canary' : '') + '.app/Contents/Resources';
```

---

It's a very hacky solution and just hijacks the PTB paths and makes them into Canary paths, but it's been working for me so far.

**Comment by BytewaveMLP** [comment]
That's odd. I tried something similar on Windows and had no success.
Perhaps I wasn't thurough enough in my modification, but everything seemed
to run fine during the patching—no errors or anything.

**Comment by ByzantineFailure** [comment]
@BytewaveMLP do you know the exact path to the Canary client on Windows?  That can help fix this.

I think what's going to happen is I'm just going to implement another flag, `--isCanary`, on the installer and check that it's not also flagged with `--isPTB` (will throw an error in that case).  After that just create the right path from there.  Need better logic for that anyway.

**Comment by 1byte2bytes** [comment]
The app.asar on my Windows laptop is at `C:\Users\sydne\AppData\Local\DiscordCanary\app-0.0.164\resources`, dunno if that's the folder you're looking for. The Canary exe is in the `app-0.0.164` folder.

Note, I do have an `app-0.0.163` folder for the old version of Discord

**Comment by ByzantineFailure** [comment]
Thanks!

Yeah, that makes a ton of sense w/ the app version folder. There's logic in the installer to find the right one.

**Comment by Shugabuga** [comment]
I am currently trying to get BPM installed on the latest Canary build. Did a recent change break anything? The code is injected (the error shows up in the first place) but fails to run. Here's the error in console.

<img width="555" src="https://user-images.githubusercontent.com/10801851/39097155-a5d45494-4626-11e8-808e-4751027931e5.png">

**Comment by ByzantineFailure** [comment]
Hmm, interesting.  Will take a look into that when I get the chance later
tonight (have a houseguest).

**Comment by Shugabuga** [comment]
Okay. 👍

**Comment by ByzantineFailure** [comment]
Fixed via 172f730df6c90977dbe38fc81550974de1f714b8
